test/dotnet-run3.Tests/GivenDotnetRun3RunsCsProj.cs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.DotNet.Tools.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.DotNet.Cli.Run3.Tests
+{
+    public class GivenDotnetRun3BuildsCsproj : TestBase
+    {
+        [Fact]
+        public void ItCanRunAMSBuildProject()
+        {
+            var testAppName = "MSBuildTestApp";
+            var testInstance = TestAssetsManager
+                .CreateTestInstance(testAppName)
+                .WithLockFiles();
+
+            var testProjectDirectory = testInstance.TestRoot;
+
+            new Build3Command()
+                .WithWorkingDirectory(testProjectDirectory)
+                .Execute()
+                .Should()
+                .Pass();
+
+            new Run3Command()
+                .WithWorkingDirectory(testProjectDirectory)
+                .ExecuteWithCapturedOutput()
+                .Should()
+                .Pass()
+                .And
+                .HaveStdOutContaining("Hello World!");
+        }
+
+        [Fact]
+        public void ItBuildsTheProjectBeforeRunning()
```

Comment:
At first I was thinking the implicit build would cover the explicit build case. Then I thought having explicit build at least covers that a bunch of magic isn't required to do the right build, so that explicit test is good.